docs/Setup.md

```diff
@@ -0,0 +1,130 @@
+   1. Quit Android Studio
+   1. Delete the invisible `.idea` in the root directory of the project.  This directory is partially ignored by Git, so deleting it will remove the files that are untracked
+   1. Restore the missing files in `.idea` folder from Git
+   1. Relaunch Android Studio
+2. Clean the individual Gradle project by running `./gradlew clean` which will purge local build outputs.
+3. Run Gradle with the argument `--rerun-tasks` which will effectively disable the build cache by re-running tasks and repopulating the cache.  E.g. `./gradlew assemble --rerun-tasks`
+4. Reboot your computer, which will ensure that Gradle and Kotlin daemons are completely killed and relaunched
+5. Delete the global Gradle cache under `~/.gradle/caches`
+6. If adding a new dependency or updating a dependency, a warning that a dependency cannot be found may indicate the Maven repository restrictions need adjusting
+
+## Gradle Tasks
+A variety of Gradle tasks are set up within the project, and these tasks are also accessible in Android Studio as run configurations.
+ * `assemble` - Compiles the SDK and demo application but does not deploy it
+ * `sdk-lib:connectedAndroidTest` - Runs the tests against the SDK
+ * `darkside-test-lib:connectedAndroidTest` - Runs the tests against the SDK which require a localhost lightwalletd server running in darkside mode
+ * `assembleAndroidTest` - Compiles the application and tests, but does not deploy the application or run the tests.  The Android Studio run configuration actually runs all of these tasks because the debug APKs are necessary to run the tests: `assembleDebug assembleZcashmainnetDebug assembleZcashtestnetDebug assembleAndroidTest`
+ * `detektAll` - Performs static analysis with Detekt
+ * `ktlintFormat` - Performs code formatting checks with ktlint
+ * `lint` - Performs static analysis with Android lint
+ * `dependencyUpdates` - Checks for available dependency updates.  It will only suggest final releases, unless a particular dependency is already using a non-final release (e.g. alpha, beta, RC).
+
+## Gradle Properties
+A variety of Gradle properties can be used to configure the build.  Most of these properties are optional and help with advanced configuration.  If you're just doing local development or making a small pull request contribution, you likely do not need to worry about these.
+
+### Debug Signing
+By default, the application is signed by the developers automatically generated debug signing key.  In a team of developers, it may be advantageous to share a debug key so that debug builds can access key-restricted services such as Firebase or Google Maps.  For such a setup, the path to a shared debug signing key can be set with the property `ZCASH_DEBUG_KEYSTORE_PATH`.
+
+### Firebase Test Lab
+This section is optional.
+
+For Continuous Integration, see [CI.md](CI.md).  The rest of this section is regarding local development.
```

Comment:
In the codebase we have it named `ci.md`. Applies to the Emulator WTF section too.